app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java

```diff
@@ -118,9 +119,9 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
 
     private void validateEmail() {
 
-        if (mEditTextEmail.getText().toString().trim().length() > 0) {
+        if (mEditTextEmail.getText().toString().trim().length() > 0 && mEditTextEmail.getText().toString().trim().matches(emailPattern)) {
```

Comment:
break this into two lines. This is what is causing your code to fail.

Reply:
Thank You @nyu-zeeshan